src/components/Identicon/StatusIcon.test.tsx

```diff
@@ -8,17 +10,42 @@ jest.mock('../../hooks/useSocksBalance', () => ({
 }))
 
 describe('StatusIcon', () => {
-  it('renders children in correct order, with no account and with socks', () => {
-    const supportedConnections = getConnections()
-    const injectedConnection = supportedConnections[1]
-    const component = render(<StatusIcon connection={injectedConnection} />)
-    expect(component.getByTestId('StatusIconRoot')).toMatchSnapshot()
+  describe('with no account', () => {
+    it('renders children in correct order, with with socks', () => {
+      const supportedConnections = getConnections()
+      const injectedConnection = supportedConnections[1]
+      const component = render(<StatusIcon connection={injectedConnection} />)
+      expect(component.getByTestId('StatusIconRoot')).toMatchSnapshot()
+    })
+
+    it('renders with showMiniIcons=false', () => {
+      const supportedConnections = getConnections()
+      const injectedConnection = supportedConnections[1]
+      const component = render(<StatusIcon connection={injectedConnection} showMiniIcons={false} />)
+      expect(component.getByTestId('StatusIconRoot').children.length).toEqual(0)
+    })
   })
 
-  it('renders with no account and showMiniIcons=false', () => {
-    const supportedConnections = getConnections()
-    const injectedConnection = supportedConnections[1]
-    const component = render(<StatusIcon connection={injectedConnection} showMiniIcons={false} />)
-    expect(component.getByTestId('StatusIconRoot').children.length).toEqual(0)
+  describe('with account', () => {
+    beforeEach(() => {
+      mocked(useWeb3React).mockReturnValue({
```

Comment:
is there a way to get `mocked` to infer `as ReturnType<typeof useWeb3React>`?

Reply:
`mocked` actually expects `ReturnType<typeof useWeb3React>`, and is "upset" because we only pass a `Partial`.
This _could_ be fixed by updating the typing of `mocked` to accept a `Partial`, but that would be a heavier lift and is out of scope here. This is - afaik - an idiomatic cast in a test file.

Reply:
Yup, like it or not, I am doing that in a few other test cases myself too. A bit verbose and strange, but gets the job done.